### PR TITLE
Update compileSdkVersion and gradle plugin

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
This change was made using AS's tool. Also updates compile sdk version to 31

Updates gradle (and other dependencies) using Android Studio's AGP tool.

This also requires updating compile sdk version to 31.

Verified the changes by running the app locally.

@gino-m  PTAL?

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,9 +18,9 @@
 
 buildscript {
     ext {
-        hiltVersion = "2.38.1"
+        hiltVersion = '2.40.1'
```
Comment (gino-m) [inline build.gradle:21]: Please change to double quotes for consistency.

**Review by gino-m — CHANGES_REQUESTED** [review]
Lgtm, just one nit.

File: build.gradle (hunk 1 above)
Comment (shobhitagarwal1612) [reply build.gradle]: Done

**Review by gino-m — APPROVED** [review]
🆙

**Pull request merged**